frontend/src/utils/index.js

```diff
@@ -704,3 +704,93 @@ export function targetText (target) {
       return undefined
   }
 }
+
+export function k8sVersionIsNotLatestPatch (kubernetesVersion, cloudProfileName) {
+  const allVersions = store.getters.kubernetesVersions(cloudProfileName)
+  return !!find(allVersions, ({ version, isPreview }) => {
+    return semver.diff(version, kubernetesVersion) === 'patch' && semver.gt(version, kubernetesVersion) && !isPreview
+  })
+}
+
+export function k8sVersionUpdatePathAvailable (kubernetesVersion, cloudProfileName) {
+  const allVersions = store.getters.kubernetesVersions(cloudProfileName)
+  if (k8sVersionIsNotLatestPatch(kubernetesVersion, cloudProfileName)) {
+    return true
+  }
+  const versionMinorVersion = semver.minor(kubernetesVersion)
+  return !!find(allVersions, ({ version, isPreview }) => {
+    return semver.minor(version) === versionMinorVersion + 1 && !isPreview
+  })
+}
+
+export function selectedImageIsNotLatest (machineImage, machineImages) {
+  const { version: testImageVersion, vendorName: testVendor } = machineImage
+
+  return !!find(machineImages, ({ version, vendorName, isPreview }) => {
+    return testVendor === vendorName && semver.gt(version, testImageVersion) && !isPreview
+  })
+}
+
+export function k8sExpirationForShoot (shootK8sVersion, shootCloudProfileName, k8sAutoPatch) {
```

Comment:
shouldn't this be named `k8sVersionExpirationForShoot`?

Reply:
ok